orttraining/orttraining/core/framework/pipeline.cc

```diff
@@ -87,7 +83,12 @@ const PipelineTask& PipelineSlot::GetFrontAction() const {
   return tasks_.front();
 }
 
-PipelineScheduler::PipelineScheduler(const int num_batches, const int num_stages) {
+PipelineScheduler::PipelineScheduler() {
+  num_stages_ = 0;
```

Comment:
Sure. I will use initialization rather than assignment. 

We can't make them const. This scheduler is a member of `TrainingSession`. Instead of ctor, we use `ConfigureForTraining` to initialize scheduler. If we change them to be const, the following line may throw in `ConfigureForTraining`.
```
pipeline_schedule_ = PipelineScheduler(...)
```

---
In reply to: [532602215](https://github.com/microsoft/onnxruntime/pull/5815#discussion_r532602215) [](ancestors = 532602215)